juju/juju
# destroy-model: destroy/release storage flags

## Description of change

Add --destroy-storage and --release-storage flags
to the destroy-model command. If the user specifies
neither, and there is persistent storage remaining,
then the destruction fails and they will be prompted
to choose one of the methods of storage removal.

The ModelManager API client and server facade have
both been updated accordingly. The facde has been
bumped to version 4. If a new client calls
destroy-model against an older server, they will be
forced to use --destroy-storage, as the old server
does not complain if there is any persistent storage
remaining. If an old client calls destroy-model on
the new controller, it will destroy the storage as
before.

## QA steps

See https://github.com/juju/juju/pull/7676, s/destroy-controller/destroy-model/g

## Documentation changes

Yes, destroy-model has grown two new flags that we'll need to document.

## Bug reference

None.
